include/accessor.h

```diff
@@ -302,17 +402,53 @@ class accessor<DataT, Dims, Mode, cl::sycl::access::target::host_buffer>
 };
 
 namespace detail {
-
-	template <typename DataT, int Dims, cl::sycl::access::mode Mode, typename... Args>
-	accessor<DataT, Dims, Mode, cl::sycl::access::target::global_buffer> make_device_accessor(Args&&... args) {
+	template <typename DataT, int Dims, cl::sycl::access_mode Mode, typename... Args>
+	accessor<DataT, Dims, Mode, cl::sycl::target::global_buffer> make_device_accessor(Args&&... args) {
 		return {std::forward<Args>(args)...};
 	}
 
-	template <typename DataT, int Dims, cl::sycl::access::mode Mode, typename... Args>
-	accessor<DataT, Dims, Mode, cl::sycl::access::target::host_buffer> make_host_accessor(Args&&... args) {
+	template <typename DataT, int Dims, cl::sycl::access_mode Mode, typename... Args>
+	accessor<DataT, Dims, Mode, cl::sycl::target::host_buffer> make_host_accessor(Args&&... args) {
 		return {std::forward<Args>(args)...};
 	}
 
+	template <typename TagT>
+	constexpr cl::sycl::access_mode deduce_access_mode() {
+		if constexpr(std::is_same_v<const TagT, decltype(cl::sycl::read_only)> || //
+		             std::is_same_v<const TagT, decltype(cl::sycl::read_only_host_task)>) {
+			return cl::sycl::access_mode::read;
+		} else if constexpr(std::is_same_v<const TagT, decltype(cl::sycl::read_write)> || //
+		                    std::is_same_v<const TagT, decltype(cl::sycl::read_write_host_task)>) {
+			return cl::sycl::access_mode::read_write;
+		} else {
+			return cl::sycl::access_mode::write;
+		}
+	}
+
+	template <typename TagT>
+	constexpr cl::sycl::access_mode deduce_access_mode_discard() {
+		if constexpr(std::is_same_v<const TagT, decltype(cl::sycl::read_only)> || //
+		             std::is_same_v<const TagT, decltype(cl::sycl::read_only_host_task)>) {
+			throw "Invalid access mode + no_init";
```

Comment:
This should be a `static_assert` instead of a `throw`

Reply:
done in new commit.